### PR TITLE
Add `Links` as header when the return type is a closed record

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

> $Subject
> Fixes : [`Fix adding links to close records #3212`](https://github.com/ballerina-platform/ballerina-standard-library/issues/3212)

## Examples

```ballerina
import ballerina/http;

type Order record {|
    string id;
    json 'order;
    decimal cost;
|};

type PaymentSlip record {|
    string id;
    decimal amount;
|};

service /restbucks on new http:Listener(9090) {

    @http:ResourceConfig {
        name: "order",
        linkedTo: [
            {name: "orders", relation: "update"},
            {name: "payment", relation: "payment"}
        ]
    }
    resource function post 'order(@http:Payload json 'order) returns Order {
        return {id: "001", 'order: 'order, cost: 49.30};
    }

    @http:ResourceConfig {
        name: "orders",
        linkedTo: [
            {name: "orders", relation: "update"},
            {name: "payment", relation: "payment"}
        ]
    }
    resource function put orders/[string id]() returns http:Ok {
        return http:OK;
    }

    @http:ResourceConfig {
        name: "payment"
    }
    resource function put payment/[string id](@http:Payload json payment)
           returns @http:Payload{mediaType: "application/restbucks.payment+json"} http:Accepted {
        PaymentSlip paymentSlip = { id: "1234", amount: 50.00};
        return { body: paymentSlip };
    }
}
```

```console
curl -v http://localhost:9090/restbucks/order -d '{"name":"latte", "quantity":2}' -H "Content-Type: application/json" 
```

```console
*   Trying 127.0.0.1:9090...
* Connected to localhost (127.0.0.1) port 9090 (#0)
> POST /restbucks/order HTTP/1.1
> Host: localhost:9090
> User-Agent: curl/7.79.1
> Accept: */*
> Content-Type: application/json
> Content-Length: 30
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 201 Created
< Link: </restbucks/orders/{id}>; rel="update"; methods=""PUT"", </restbucks/payment/{id}>; rel="payment"; methods=""PUT""; types=""application/restbucks.payment+json""
< content-type: application/json
< content-length: 66
< server: ballerina
< date: Thu, 4 Aug 2022 15:18:16 +0530
< 
* Connection #0 to host localhost left intact
{"id":"001", "order":{"name":"latte", "quantity":2}, "cost":49.30}
```

```console
curl -v http://localhost:9090/restbucks/payment/001 -d '{"amount":50.00}' -H "Content-Type: application/json" -X PUT
```

```console
*   Trying 127.0.0.1:9090...
* Connected to localhost (127.0.0.1) port 9090 (#0)
> PUT /restbucks/payment/001 HTTP/1.1
> Host: localhost:9090
> User-Agent: curl/7.79.1
> Accept: */*
> Content-Type: application/json
> Content-Length: 16
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 202 Accepted
< Link: </restbucks/order>; rel="order"; methods=""POST""
< content-type: application/restbucks.payment+json
< content-length: 29
< server: ballerina
< date: Thu, 4 Aug 2022 15:23:09 +0530
< 
* Connection #0 to host localhost left intact
{"id":"1234", "amount":50.00}
```

## Checklist
- [x] Linked to an issue
- [ ] <s>Updated the changelog</s>
- [x] Added tests
- [x] Updated the spec

### CODE REVIEW

File: ballerina/http_connection.bal [hunk 1]
```diff
@@ -292,7 +291,28 @@ isolated function arrayToString(string[] arr) returns string {
     return arrayString.substring(1, arrayString.length() - 1);
 }
 
-isolated function setPayload(anydata payload, Response response, string? mediaType = (), boolean setETag = false) {
+isolated function addLinksToPayload(anydata message, map<Link>? links) returns [boolean, anydata] {
+    if links !is () {
```
Comment (ayeshLK) [inline ballerina/http_connection.bal:295]:
```suggestion
    if links is map<Link> {
```

Comment (TharmiganK) [reply ballerina/http_connection.bal:295]: I am creating and returning this map from the native side (HttpCallableUnitCallback). And when we check this as `if links is map<Link>` it returns false. Not sure why

Comment (chamil321) [reply ballerina/http_connection.bal:295]: Does it work for `map<json>`?

**Review by chamil321 — APPROVED** [review]

File: ballerina/http_connection.bal (hunk 1 above)
Comment (TharmiganK) [reply ballerina/http_connection.bal]: No, not worked for `map<json>` or `json`. There was a bug when creating the `BMap` and fixed via [7643324](https://github.com/ballerina-platform/module-ballerina-http/pull/1167/commits/7643324a218e8f51fd493139840d9752a06a7555)

**Pull request merged**